### ISSUE TITLE
reading pp files with non-gregorian calendars does not work

### ISSUE THREAD

**Issue by bnlawrence** [NCAS-CMS/cf-python#170]
I am attempting to aggregate as set of UM pp files in a directory.
These include a file which appears to have a 30th of February.
I had expected cf-python to warn me and change to a `360_day` calendar
(or allow me to specify it in the read statement).

Error message
```
 File "cftime/_cftime.pyx", line 948, in cftime._cftime.datetime.__init__
  File "cftime/_cftime.pyx", line 1492, in cftime._cftime.assert_valid_date
ValueError: invalid day number provided in cftime.datetime(1992, 2, 30, 0, 0, 0, 0, calendar='gregorian')
```

Environment: Sci5, private conda install:
```
Platform: Linux-3.10.0-1127.19.1.el7.x86_64-x86_64-with-centos-7.8.2003-Core
HDF5 library: 1.10.6
netcdf library: 4.7.4
udunits2 library: /home/users/lawrence/miniconda3/envs/macbook/lib/libudunits2.so.0
python: 3.7.9
netCDF4: 1.5.3
cftime: 1.3.0
numpy: 1.19.4
psutil: 5.7.3
scipy: 1.5.3
matplotlib: 3.3.3
ESMF: not available
cfdm: 1.7.11
cfunits: 3.2.4
cfplot: 3.0.38
cf: 3.1.0

```

**Comment by davidhassell** [comment]
Hi Bryan, Could you let me know what the LBTIM header value is? The code is beholden to that - I'm most interested in the "units" part of the number (https://github.com/NCAS-CMS/cf-python/blob/master/cf/read_write/um/umread.py#L550-L558)

Thanks.

**Comment by davidhassell** [comment]
Hi Bryan - I think I may have worked this out - the default calendar behaviour of `cftime` changed at `cftime 1.3.0`. Assuming thaet this is the case, you either go back to `cftime 1.2.0`, or wait until we release cf-python 3.8.0 later this week. The new cf-python requires `cftime>=1.3.0` and is adjusted accordinginly.

**Comment by bnlawrence** [comment]
You're too quick for me ... does this mean I don't need to find the LBTIM header? If so, I can wait for cf-python 3.8.0.
Ta.

**Comment by davidhassell** [comment]
Yes - no need to go digging in the header, I think. It would be nice to test this theory out, though ... do you have a test file handy? Thanks.
